src/main/java/duke/operation/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package duke.operation;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * This is the Deadline class for deadline tasks.
+ */
+public class Deadline extends Task {
+	protected LocalDateTime by;
+
+	/**
+	 * Constructor for Deadline objects.
+	 *
+	 * @param description string input
+	 * @param by          deadline date
+	 */
+	public Deadline(String description, LocalDateTime by) {
+		super(description);
+		this.by = by;
+	}
+
+	/**
+	 * Splits deadline input string into body and deadline date.
+	 *
+	 * @param input input string
+	 * @return new split Deadline object
+	 */
+	public static Deadline splitDeadline(String input) {
+		String[] partsOfDeadline = input.split("/by ");
+		String deadlineContent = partsOfDeadline[0].substring(9);
+		DateTimeFormatter dateTimeFormatterFrom = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+		LocalDateTime by = LocalDateTime.parse(partsOfDeadline[1], dateTimeFormatterFrom);
+		return new Deadline(deadlineContent, by);
+	}
+
+	@Override
+	public String toString() {
+		DateTimeFormatter dateTimeFormatterTo = DateTimeFormatter.ofPattern("MMM-dd-yyyy HH:mm");
+		return "[D]"
+				+ super.toString()
+				+ " (by: "
+				+ this.by.format(dateTimeFormatterTo)
+				+ ")";
```

Comment:
I like the way you format this such that it is very clean and easy to read but I am not sure if this fits the module's coding standards :0

Reply:
Thanks for reminding and I'll take a look at the coding standards again:)

Reply:
The code format is shown differently on GitHub vs IDE. Not sure why this happens.